### PR TITLE
Variance as a Metric for Active Learning

### PR DESCRIPTION
**Pull request by finalelement**
Signed-off-by: Vishwesh Nath <vnath@nvidia.com>

### Description
Add a metric function which will return either 2D/3D spatial map depending upon input's dimensions or a single scalar value of mean/sum of variance given user's choice

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/metrics/active_learning_metrics.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            y_pred=y_pred,
+            include_background=self.include_background,
+            spatial_map=self.spatial_map,
+            scalar_reduction=self.scalar_reduction,
+            threshold=self.threshold,
+        )
+
+
+def compute_variance(
+    y_pred: torch.Tensor, include_background=True, spatial_map=False, scalar_reduction="mean", threshold=0.0005
```
Comment (Nic-Ma) [inline monai/metrics/active_learning_metrics.py:68]: Please add the typehints for these args.

Thanks.

**Review by Nic-Ma — COMMENTED** [review]
Looks good to me.
Please add the new metric to the doc:
https://github.com/Project-MONAI/MONAI/blob/dev/docs/source/metrics.rst

Thanks.

File: monai/metrics/active_learning_metrics.py (hunk 1 above)
Comment (finalelement) [reply monai/metrics/active_learning_metrics.py:68]: Thank you will add metric to doc and add typehints @Nic-Ma

File: monai/metrics/active_learning_metrics.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        y_pred, y = ignore_background(y_pred=y_pred, y=y)
+
+    # Set any values below 0 to threshold
+    y_pred[y_pred <= 0] = threshold
+
+    n_len = len(y_pred.shape)
+
+    if n_len < 4 and spatial_map is True:
+        print(f"Spatial Map set to True for a 1-D array, the provided input is {y_pred.shape}")
+        print("Spatial map requires a 2D/3D image with N-repeats and C-channels")
```
Comment (wyli) [inline monai/metrics/active_learning_metrics.py:97]:
```suggestion
        warnings.warn(f"Spatial Map set to True for a 1-D array, the provided input is {y_pred.shape}",
                       "Spatial map requires a 2D/3D image with N-repeats and C-channels")
```

if you add this change, `import warnings` is needed in this file

**Review by wyli — APPROVED** [review]

File: monai/metrics/active_learning_metrics.py (hunk 1 above)
Comment (finalelement) [reply monai/metrics/active_learning_metrics.py]: All added & addressed

**Pull request merged**